Check for duplicate device before adding to resolving list

ibaz-nordic opened the pull request:
In Core spec v5.3, controller has the option to accept or reject when
    the same device is added to RL.  This check ensures duplicate device
    will not be sent to the controller.

Commit:
modem: choice symbol for UART backend

Add a choice symbol that is used to select which UART backend to use.
This allows backends that don't use the interrupt API to be implemented.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

Commit:
modem: modem_iface_uart_async: added

Adds a communications backend based on the asynchronous UART API,
instead of the interrupt-driven UART API. The primary advantage of this
backend is an improved robustness to dropping bytes under high interrupt
or critical section loads.

Under all loads system efficiency is improved by:
 * Reducing the time spent writing out individual bytes.
 * Reducing the number of UART interrupts fired.
 * Waking up the RX thread much less often.

When utilising this backend over `nordic,nrf-uarte` on a nRF52840, the
baudrate of an esp-at modem could be pushed to at least 921600 without
dropping bytes, compared to a maximum of 230400 with the interrupt API.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

Commit:
serial: test: update selected symbols

Pretend that the serial test driver supports the interrupt and async
API's, as these can be required for various drivers. Also select
`SERIAL_HAS_DRIVER` so that the serial library will be included.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

Commit:
tests: build_all: modem: remove custom serial

Remove a custom Kconfig symbol that was used to indicate serial
interrupt support, as the test serial driver now selects the appropriate
symbols.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

Commit:
tests: build_all: modem: test esp-at

Add build tests for the `espressif,esp-at` driver, together with both
the interrupt and async variants of modem_ifact_uart.

Only build the async API variant for emulated platforms, as many
platforms do not build cleanly when `UART_ASYNC_API` is enabled without
custom setup of DMA nodes.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

Commit:
wifi: esp_at: fix pointer logging

Use the proper `%p` printf specifier when printing memory addresses,
instead of casting to an integer, which may not be the same size as a
pointer.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

Commit:
serial: litex: remove irrational dependency

Depending on `!SERIAL_SUPPORT_INTERRUPT` to enable the driver does not
make any sense, as this is a symbol selected by drivers to signify that
they support interrupts. Simply not selecting this symbol is enough to
convey the desired intention.

This fixes Kconfig problems when the driver is compiled together with
a dummy serial driver which does select `SERIAL_SUPPORT_INTERRUPT`.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

dcpleung committed:
scripts: logging/dictionary/sys-t: log module info in collateral

This updates the MIPI Sys-T collateral generation to include log
modules in the XML file. This allows the parser to show where
each log message is coming from.

Note that the addition of Name propety in sys:Client is to clarify
that this collateral is from Zephyr.

Also note he addition of Mask property in syst:Guid is for parser
to match this collateral to any incoming messages, as the module
IDs are encoded in the set bits in the GUID. Without the mask,
the parser will discard any messages with non-zero module IDs.
This is based on the observation of the reference parser,
systprint, where pseudo GUIDs are generated from the module/unit
IDs (origin unit) in the incoming messages for matching with
collaterals.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
logging: syst: embed log message source IDs in Sys-T messages

This embeds the log message source IDs inside the origin unit
as module IDs in Sys-T messages. This allows Sys-T message
parsers to see where the log messages are coming from.

This is enabled by default if using Sys-T catalog messages as
the collateral XML file contains the information to interpret
the module ID.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

ibaz-nordic committed:
bluetooth: host: Check for duplicate device before adding to resolving list

In Core spec v5.3, controller has the option to accept or reject when
the same device is added to RL.  This check ensures duplicate device
will not be sent to the controller.

Signed-off-by: Azizah Ibrahim <azizah.ibrahim@nordicsemi.no>

ibaz-nordic committed:
bluetooth: host: Check for duplicate device before adding to resolving list

In Core spec v5.3, controller has the option to accept or reject when
the same device is added to RL.  This check ensures duplicate device
will not be sent to the controller.

Signed-off-by: Azizah Ibrahim <azizah.ibrahim@nordicsemi.no>

jori-nordic commented on subsys/bluetooth/host/id.c:
```diff
@@ -773,13 +773,23 @@ void bt_id_add(struct bt_keys *keys)
+	 * If the command is rejected then the error code
+	 * Invalid HCI Command Parameters (0x12) should be used.
+	 */
+	if (!duplicate_address) {
```
can we move all the checking logic here instead ? I think the code flow will be much clearer.

aasthagr committed:
MAINTAINERS: Add Collaborator for Logging subsystem.

Adds @aasthagr as a Collaborator in Logging.

Signed-off-by: Aastha Grover <aastha.grover@intel.com>

MaureenHelm committed:
dts: x86: intel: Move SoC devicetree includes under a vendor directory

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
dts: arm64: intel: Move SoC devicetree includes under a vendor directory

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
dts: nios2: intel: Move SoC devicetree includes under a vendor directory

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
dts: riscv: ite: Move SoC devicetree includes under a vendor directory

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
dts: riscv: andes: Move SoC devicetree includes under a vendor directory

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
dts: riscv: microsemi: Move SoC devicetree includes under a vendor dir

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
dts: riscv: sifive: Move SoC devicetree includes under a vendor dir

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
dts: riscv: openisa: Move SoC devicetree includes under a vendor dir

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
dts: riscv: telink: Move SoC devicetree includes under a vendor dir

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
dts: sparc: gaisler: Move SoC devicetree includes under a vendor dir

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
dts: arc: synopsys: Move SoC devicetree includes under a vendor dir

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
dts: arm64: fvp: Move SoC devicetree includes under a vendor directory

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
dts: arm64: qemu: Move SoC devicetree includes under a vendor directory

Cleans up SoC devicetree include file locations to follow the convention
of dts/<arch>/<vendor>/

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

quic-egmc committed:
arch/arm64: update gicv3 sre enablement

Fix writing of ICC_SRE_EL3 to or-in bits to align
with original intent to read-modify-write this
register.

Also disable FIQ and IRQ bypass so interrupt delivery
occurs through GIC.  Platforms may choose to override
this behavior in z_arm64_el3_plat_init implementations.

Remove ICC_SRE_EL3 config from viper and qemu since
this is now handled in the arm64 arch core.

Signed-off-by: Eugene Cohen <quic_egmc@quicinc.com>

GeorgeCGV committed:
drivers: uart_stm32: fix device is ready for tx dma in async init

Fixes TX DMA device is ready check in async initialization function.

Signed-off-by: Georgij Cernysiov <geo.cgv@gmail.com>

henrikbrixandersen committed:
tests: canbus: isotp: conformance: remove unused random_data_len variable

Remove unused random_data_len variable.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
tests: canbus: isotp: implementation: remove random_data_len variable

Do not rely on hardcoded array size. Use sizeof() instead.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
tests: canbus: isotp: implementation: allow data to propagate

Allow transmitted data to propagate in test_bind_unbind().

Fixes: #44557

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

tristan-google committed:
twister: Fix dependency on west when checking for projects

Twister should be partially usable even without West installed, however,
Twister unconditionally tries to import it when initializing to check
for west projects and crashes if it is not there.

This PR changes the behavior to return None if the west imports failed,
which appears to be more in line with what was intended. This allows
Twister to proceed and successfully run tests in a non-west environment.
There is no impact on environments that do have west installed.

Fixes #45355

Signed-off-by: Tristan Honscheid <honscheid@google.com>

hunterhu3000 committed:
test: Remove usec test for pwm_api

As the pwm interface only takes nanosecond for parameter,
the original test case for mircosecond is not needed.
Check github issue 44887 for more info.

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

rettichschnidi committed:
doc: k_sleep does not take raw integers

Using k_msleep seems easier than k_sleep(K_MSEC(1000));

Signed-off-by: Reto Schneider <reto.schneider@husqvarnagroup.com>

nashif committed:
Revert "ci: run twister with CONFIG_LEGACY_INCLUDE_PATH=n"

This reverts commit e4ebba8025978861a694a22e505af758ace15dd9.

This is still not ready. We have lots of usages of old include paths.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

ibaz-nordic committed:
Merge branch 'rl_check_add' of https://github.com/ibaz-nordic/zephyr into rl_check_add

ibaz-nordic commented:
I made a mess
